src/utils/apollo-client.ts

```diff
@@ -0,0 +1,49 @@
+import {
+    ApolloClient,
+    InMemoryCache,
+    NormalizedCacheObject,
+    ApolloQueryResult,
+    gql,
+} from '@apollo/client';
+import { RestLink } from 'apollo-link-rest';
+
+import { Article } from '../interfaces/article';
+
+class ApolloGraphQLClient {
+    _client: ApolloClient<NormalizedCacheObject>;
+
+    constructor() {
+        this._client = new ApolloClient({
+            cache: new InMemoryCache(),
+            uri: `${process.env.STRAPI_URL}/graphql`,
+        });
+    }
+}
+class ApolloRestClient {
+    _client: ApolloClient<NormalizedCacheObject>;
+
+    constructor() {
+        this._client = new ApolloClient({
+            cache: new InMemoryCache(),
+            link: new RestLink({ uri: process.env.STRAPI_URL }),
+        });
+    }
+
+    getArticles = async (): Promise<Article[]> => {
```

Comment:
getArticles should not be a client function.

Reply:
Where should it go? My thought was that it should be attached to the instance of the class in order to reuse the underlying apollo client, so we can utilize the cache and don't have to instantiate a new one every time we make a request.

Reply:
it should be under a different folder api or data. getArticles primary job will be to construct the query, pass it to client instance. Client's job will be to run the query passed.

eg : getArticles = (client) => {
const query = ``;
const data = await client.query({query});
return data;
}

while testing you can 
mock client instance and mock await client.query call. 

about this part - we can utilize the cache and don't have to instantiate a new one every time we make a request - i think it is still valid in my case, I'm still using the same instance of the client. 

All said I'm okay with ever call you take.

Reply:
Ok that makes sense to me. I've now separated out `getArticles` to it's own location (`src/requests/get-articles.ts`) where it accepts a client. Now when we get the static props, we just instantiate the client and call `getArticles` with the client. 

I've also added a unit test to this. It took me a while to get the mocking right, but it seems to be working now.